lib/mxlib/matrix.js

```diff
@@ -25,36 +24,12 @@ var mkUserId = function(localpart) {
     return "@" + localpart + ":" + hsDomain;
 };
 
-var getAliasLocalpart = function(alias) {
-    return alias.split(":")[0].substring(1);
-};
-
-module.exports.getAppServiceUserId = function() {
-    return appServiceUserId;
-};
-
 module.exports.setMatrixClientConfig = function(config, bridge) {
     hsDomain = config.domain;
-    appServiceUserId = "@" + config.localpart + ":" + hsDomain;
     globalBridge = bridge;
 };
 
-module.exports.joinMappedRooms = function() {
-    var d = promiseutil.defer();
-    var promises = [];
-    store.rooms.getRoomIdsFromConfig().done(function(roomIds) {
-        var lib = module.exports.getMatrixLibFor();
-        roomIds.forEach(function(roomId) {
-            promises.push(lib.joinRoom(roomId));
-        });
-        promiseutil.allSettled(promises).then(function() {
-            d.resolve();
-        });
-    });
-    return d.promise;
-};
-
-module.exports.decodeMxc = function(mxcUri) {
+module.exports.decodeMxc = function(mxcUri) { // actions.js (de-singleton) TODO
```

Comment:
`actions.js` needs to have extra context of the HS URL so it can convert MXC URIs to HTTP URIs correctly. This could be done either by instantiating actions with some surrounding context, or explicitly passing the HS URL around the place when creating messages.